### PR TITLE
vs-server: Enable https connecting to vs when AuthenticationCertificate is provided.

### PR DESCRIPTION
**Pull request by weikanglim**
Enable https connecting to vs when AuthenticationCertificate is provided in vs-server.

Also, allow AZD_AUTH_CERT to inherit the same behavior when interacting with azd CLI.

### CODE REVIEW

File: cli/azd/cmd/container.go [hunk 1]
```diff
@@ -523,11 +528,53 @@ func registerCommonDependencies(container *ioc.NestedContainer) {
...
+			}
+
+			caCertPool := x509.NewCertPool()
+			caCertPool.AddCert(cert)
+			tlsConfig := &tls.Config{
+				RootCAs:    caCertPool,
+				MinVersion: tls.VersionTLS12,
+			}
+
+			transport := http.DefaultTransport.(*http.Transport).Clone()
```
Comment (ellismg) [inline cli/azd/cmd/container.go:557]: Just wondering - why this vs just `&http.Transport{TLSClientConfig: tlsConfig}`.  I assume that the DefaultTransport has other state we want to inherit/share but I wonder what that is.

**Review by ellismg — APPROVED** [review]
LGTM!

I did wonder about trying to share the "create the correct client given a certificate (and perhaps a transport, and when nil we use the DefaultTransport" but I wasn't really sure where we'd put it, and there's so little code here that I'm not worried about the duplication.

File: cli/azd/cmd/container.go (hunk 1 above)
Comment (weikanglim) [reply cli/azd/cmd/container.go:557]: The connection timeouts and number of idled connections are all useful and more sane defaults than when not provided. [k8s](https://github.com/kubernetes/kubernetes/blob/b722d017a34b300a2284b890448e5a605f21d01e/staging/src/k8s.io/apimachinery/pkg/util/net/http.go#L105C5-L105C22) does something similar when overriding TLS client configs.

**Comment by weikanglim** [comment]
Moved to #3712

**Pull request closed without merging**